collector/pg_total_relation_size.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const totalRelationSizeSubsystem = "total_relation_size"
+
+func init() {
+	registerCollector(totalRelationSizeSubsystem, defaultDisabled, NewPGTotalRelationSizeCollector)
+}
+
+type PGTotalRelationSizeCollector struct {
+	log log.Logger
+}
+
+func NewPGTotalRelationSizeCollector(config collectorConfig) (Collector, error) {
+	return &PGTotalRelationSizeCollector{log: config.logger}, nil
+}
+
+var (
+	totalRelationSizeBytes = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, totalRelationSizeSubsystem, "bytes"),
+		"total disk space usage for the specified table and associated indexes",
+		[]string{"schemaname", "relname"},
+		prometheus.Labels{},
+	)
+
+	totalRelationSizeQuery = `
+		SELECT
+			relnamespace::regnamespace as schemaname,
+			relname as relname,
+			pg_total_relation_size(oid) bytes
+		FROM pg_class
```

Comment:
It seems like there are several collectors that query `pg_class`. I wonder if we should refactor this into a `pg_class` collector that gathers all the data in one shot.